plugin/trino-base-jdbc/src/main/antlr4/io/trino/plugin/jdbc/expression/ConnectorExpressionPattern.g4

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+grammar ConnectorExpressionPattern;
+
+tokens {
+    DELIMITER
+}
+
+standaloneExpression
+    : expression EOF
+    ;
+
+standaloneType
+    : type EOF
+    ;
+
+expression
+    : call
+    | expressionCapture
+    ;
+
+call
+    : identifier '(' expression (',' expression)* ')' (':' type)?
+    ;
+
+expressionCapture
+    : identifier ':' type
+    ;
+
+type
+    : identifier
+    | identifier '(' typeParameter (',' typeParameter)* ')'
+    ;
+
+typeParameter
+    : number
+    | identifier
+    ;
+
+identifier
+    : IDENTIFIER
+    ;
+
+number
+    : INTEGER_VALUE
+    ;
+
+IDENTIFIER
+    : (LETTER | '_' | '$') (LETTER | DIGIT | '_' | '$')*
```

Comment:
Not that these are not SQL identifiers (no delimited vs non-delimited), so they should be treated as already-canonicalized. Also, that implies that the character set is potentially anything that can go inside a delimited SQL identifier (i.e., any UTF-8 character). This is going to be hard to do without introducing some kind of escaping for whitespace or other characters.

Concretely, if a SQL query does something like:

```
SELECT "123 some function"(a, b, c) FROM t
```

(i.e., it calls a function named `123 some function`)

That name needs to be representable in this grammar. The following pattern would obviously not work due to 1) whitespace 2) leading numbers in the function name:

`123 some function(a : varchar, b : varchar, c : varchar) : bigint`

Reply:
> Note that these are not SQL identifiers (no delimited vs non-delimited)

yes, they aren't

>. so they should be treated as already-canonicalized. 

in a sense of lower/upper-case? 
yes, it should be case-sensitive

> Also, that implies that the character set is potentially anything that can go inside a delimited SQL identifier (i.e., any UTF-8 character). This is going to be hard to do without introducing some kind of escaping for whitespace or other characters.

For such a case we will introduce quotes to allow a name be anything.
We don't have such need yet, but once we have, we'll do it.

Reply:
note that this is convenience syntax to simplify most of the cases. It doesn't need to support all the cases. For example, it doesn't support varargs today. It can be made to support varargs, and we may or may not choose to add that, depending on demand. "99%" cases are function names that are ordinary words, with fixed number of arguments, potentially with overloads for different types.